spanner/value.go

```diff
@@ -3620,30 +3766,42 @@ func encodeValue(v interface{}) (*proto3.Value, *sppb.Type, error) {
 	case []GenericColumnValue:
 		return nil, nil, errEncoderUnsupportedType(v)
 	case protoreflect.Enum:
-		var protoEnumfqn string
 		if v != nil {
+			var protoEnumfqn string
 			rv := reflect.ValueOf(v)
 			if rv.Kind() != reflect.Ptr || !rv.IsNil() {
 				pb.Kind = stringKind(strconv.FormatInt(int64(v.Number()), 10))
 				protoEnumfqn = string(v.Descriptor().FullName())
 			} else {
-				return nil, nil, errNilSrcPtr(v)
+				defaultType := reflect.Zero(rv.Type().Elem()).Interface().(protoreflect.Enum)
+				protoEnumfqn = string(defaultType.Descriptor().FullName())
 			}
+			pt = protoEnumType(protoEnumfqn)
+		}
+	case NullEnum:
+		if v.Valid {
+			return encodeValue(v.EnumVal)
+		} else {
+			return nil, nil, errNotValidSrc(v)
 		}
-		pt = protoEnumType(protoEnumfqn)
 	case proto.Message:
-		var protoMessagefqn string
-		if v != nil && proto.MessageReflect(v).IsValid() {
-			bytes, err := proto.Marshal(v)
-			if err != nil {
-				return nil, nil, err
+		if v != nil {
+			if proto.MessageReflect(v).IsValid() {
+				bytes, err := proto.Marshal(v)
+				if err != nil {
+					return nil, nil, err
+				}
+				pb.Kind = stringKind(base64.StdEncoding.EncodeToString(bytes))
 			}
-			pb.Kind = stringKind(base64.StdEncoding.EncodeToString(bytes))
-			protoMessagefqn = string(proto.MessageReflect(v).Descriptor().FullName())
+			protoMessagefqn := string(proto.MessageReflect(v).Descriptor().FullName())
+			pt = protoMessageType(protoMessagefqn)
+		}
```

Comment:
When would v be nil? Is doing nothing expected?

If it's expected, please add a comment.

Reply:
> When would v be nil? Is doing nothing expected?

v is `nil` only when a untyped null value is passed by user. As pb and pt are assigned by default in the beginning of switch case there is nothing to be done.